invokeai/app/invocations/generate.py

```diff
@@ -6,22 +6,40 @@
 import numpy as np
 from torch import Tensor
 
-from pydantic import Field
+from pydantic import BaseModel, Field
+from invokeai.app.invocations.models.config import (
+    InvocationConfig,
+)
 
 from invokeai.app.models.image import ImageField, ImageType
+from invokeai.app.invocations.util.get_model import choose_model
 from .baseinvocation import BaseInvocation, InvocationContext
 from .image import ImageOutput
 from ...backend.generator import Txt2Img, Img2Img, Inpaint, InvokeAIGenerator
 from ...backend.stable_diffusion import PipelineIntermediateState
 from ..models.exceptions import CanceledException
 from ..util.step_callback import diffusers_step_callback_adapter
 
-SAMPLER_NAME_VALUES = Literal[
-    tuple(InvokeAIGenerator.schedulers())
-]
+SAMPLER_NAME_VALUES = Literal[tuple(InvokeAIGenerator.schedulers())]
+
+
+class SDImageInvocation(BaseModel):
+    """Helper class to provide all Stable Diffusion raster image invocations with additional config"""
+
+    # Schema customisation
+    class Config(InvocationConfig):
+        schema_extra = {
+            "ui": {
+                "tags": ["stable-diffusion", "image"],
+                "type_hints": {
+                    "model": "model",
```

Comment:
The one worry I have with this is that it's pretty hidden/separate from the fields below. Someone adding a `model` field to some other invocation may not realize they need to also be adding this to their config.

Reply:
Yeah I agree. For now, this works, but maybe a python person can figure out a better solution. The main thing this PR introduces is the structure of the UI hints, with the implementation kinda just being what I could figure out.